server/common/utils/helpers.js

```diff
@@ -60,25 +61,34 @@ const responseWithList = (list, userId) => {
 };
 
 const responseWithListsMetaData = (lists, userId) =>
-  _map(lists, list => {
-    const { cohortId, favIds, items, ...rest } = list;
-    const activeItems = items.filter(item => !item.isArchived);
-    const doneItemsCount = activeItems.filter(item => item.isOrdered).length;
-    const unhandledItemsCount = activeItems.length - doneItemsCount;
-
-    const listToSend = {
-      ...rest,
-      doneItemsCount,
-      isFavourite: checkIfArrayContainsUserId(favIds, userId),
-      unhandledItemsCount
-    };
-
-    if (cohortId) {
-      listToSend.cohortId = cohortId;
-    }
-
-    return listToSend;
-  });
+  _filter(
+    _map(lists, list => {
+      const { cohortId: cohort, favIds, items, ...rest } = list;
+
+      if (cohort && !cohort.isArchived) {
+        return;
+      }
+
+      const activeItems = items.filter(item => !item.isArchived);
+      const doneItemsCount = activeItems.filter(item => item.isOrdered).length;
+      const unhandledItemsCount = activeItems.length - doneItemsCount;
+      const cohortId = cohort ? cohort._id : null;
```

Comment:
do you need to do this?
if cohortId will be undefined or null it will be falsy, if there will be a value it will be truthy so it will work the same way